### PR TITLE
Parameterize prefix mask call (needed by POD-Attention)

### PR DESCRIPTION
**Pull request by AKKamath**
Passes the threadId to the prefix mask call manually. Shouldn't change existing code, but is necessary since POD remaps threadIds and blockIds.

### CODE REVIEW

File: benchmarks/bench_mixed_attention.py [hunk 1]
```diff
@@ -93,6 +97,18 @@ def run_bench(
...
+            q_p,
+            k_p,
+            v_p,
+            q_d,
+            kv_data,
+            causal_p=causal,
+        )
+        o_pod = torch.cat([o_d, o_p], dim=0)
+        # Verify output matches
+        if not torch.allclose(o, o_pod, rtol=1e-3, atol=1e-3):
```
Comment (yzh119) [inline benchmarks/bench_mixed_attention.py]: You can use [torch.testing.assert_close](https://docs.pytorch.org/docs/stable/testing.html#torch.testing.assert_close) which is simpler.

File: benchmarks/bench_mixed_attention.py [hunk 2]
```diff
@@ -76,15 +77,18 @@ def run_bench(
         q_p = q[d_q_indptr[-1] :]
         k_p, v_p = kv_data[d_kv_indptr[-1] :].unbind(1)
         k_p, v_p = k_p.squeeze(1), v_p.squeeze(1)
+        kv_indices_d = torch.arange(
+            0, d_kv_indptr[-1], device="cuda:0", dtype=torch.int32
```
Comment (Edenzzzz) [inline benchmarks/bench_mixed_attention.py]: `device=device`?

File: benchmarks/bench_mixed_attention.py (hunk 1 above)
Comment (AKKamath) [reply benchmarks/bench_mixed_attention.py]: Thanks, fixed.

File: benchmarks/bench_mixed_attention.py (hunk 2 above)
Comment (AKKamath) [reply benchmarks/bench_mixed_attention.py]: Fixed

**Review by yzh119 — APPROVED** [review]

**Comment by Edenzzzz** [comment]
I will try to add BatchedPrefill for POD in the meantime. It seems mostly about setting up the params and page indices in `wrapper.plan` and `pod_with_kv_cache_tensor`
